osu.Game/Rulesets/Scoring/HitWindows.cs

```diff
@@ -174,14 +196,35 @@ public double WindowFor(HitResult result)
         /// </summary>
         /// <param name="timeOffset">The time offset.</param>
         /// <returns>Whether the <see cref="HitObject"/> can be hit at any point in the future from this time offset.</returns>
-        public bool CanBeHit(double timeOffset) => timeOffset <= WindowFor(LowestSuccessfulHitResult());
+        public bool CanBeHit(double timeOffset) => Contains(timeOffset, LowestSuccessfulHitResult());
 
         /// <summary>
         /// Retrieve a valid list of <see cref="DifficultyRange"/>s representing hit windows.
         /// Defaults are provided but can be overridden to customise for a ruleset.
         /// </summary>
         protected virtual DifficultyRange[] GetRanges() => base_ranges;
 
+        public class LegacyHitWindows : HitWindows
```

Comment:
I don't really agree with this argument. I used this word because standard, taiko, catch and mania are called legacy rulesets in the codebase:
https://github.com/ppy/osu/blob/a4c9e9f84dfe8d37122bb4933ea58b47a9093af9/osu.Game/Extensions/ModelExtensions.cs#L76
And some kind of word is necessary (at least with the new class approach), because these hit windows would not be recommended to be used for new custom rulesets, and we're only keeping them to not diverge from stable. If you can think of a better name, feel free to recommend it (i also used `HistoricalHitWindows` before commiting).